### ISSUE TITLE
Provided WASM is 2x the size it once was

### ISSUE THREAD

**Issue by polarathene** [BinomialLLC/basis_universal#133]
From [this commit last year](https://github.com/BinomialLLC/basis_universal/commit/2cb6bdadc1214c706b933ed0d4e4b6adbe23b662#diff-69fe6bc9448e5e5ba0158f3a264935e3) WASM builds included BC7 support and any other changes since then. 

While the WASM size is currently 437KB, that's stillmore than double what a more minimal build offered previously. Unsure of the compressed(gzip) size is, combined with the JS loader back then it was mentioned as 125KB.

Is most of the additional size from BC7 still, or has newer features since such as UASTC also bulked it up notably? Is it enough of a difference to warrant a slimmer variant, or should those who want that build the WASM themselves?

**Comment by richgel999** [comment]
Yes - we're going to work on reducing this. I have a custom std::vector replacement which should help reduce the compiled size. This work will commence once we finish the standards documents, which are coming along.

**Comment by polarathene** [comment]
A minimal common case WASM which keeps file size low while supporting the relevant transcode targets developers are likely wanting to use would be great, but perhaps there is value in supplying a kitchen sink variant for testing the remaining transcode targets easily, and if there was any need for a different build in production for that developers needs, they'd opt for a custom build themselves at that point. Minimizing friction for adoption would be good though :)

Google has on their Draco compression repo, [a recommendation to load the dependency from a common CDN link](https://github.com/google/draco#version-136-release)(gstatic) to leverage cache hits as adoption grows. Would similar be beneficial to Basis?(I suppose on the web it may be more common for ThreeJS and BabylonJS to provide their own)

It may serve as one other approach to potentially reducing the WASM size impact. Since Google has sponsored Basis development, a similar gstatic resource could perhaps be viable?

**Comment by richgel999** [comment]
The std::vector replacement is in. Leaving this open for investigation.

**Comment by richgel999** [comment]
Note I have removed a bunch of unused/dead code from the v1.16 transcoder, which is going out in a week or so. This should make a dent on the transcoder's WASM size.
